packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -0,0 +1,562 @@
+  /// given layer links if they are not null. It is possible only one layer is
+  /// non-null if this handler is only responsible for painting that non-null
+  /// layer link.
+  ///
+  /// If you are implementing this method as a result of mixing [Selectable] to
+  /// a leaf render object, consider also mixing the [HandleLayersPainterMixin]
+  /// to get the default implementation.
+  ///
+  /// See also:
+  ///  * [HandleLayersPainterMixin]: which paint layer links automatically for
+  ///    the mixer.
+  void pushHandleLayers(LayerLink? startHandle, LayerLink? endHandle);
+
+  /// Gets the selected content from this selectable.
+  ///
+  /// Return `null` if nothing is selected.
+  Object? getSelectedContent();
+
+  /// handles the [SelectionEvent] sent to this object.
+  ///
+  /// The subclasses need to update its selection or delegate the selection
+  /// event to its subtree.
+  ///
+  /// The [event] is a subclass of [SelectionEvent]. Use runtime type check to
+  /// determine what kind of event is dispatched to this handler and handle it
+  /// accordingly.
+  SelectionResult dispatchSelectionEvent(SelectionEvent event);
+}
+
+/// An object that can be selected by the [SelectionArea] widget.
+///
+/// This object receives selection events and is responsible to draw selection
+/// highlight. In order to receive the selection event, the subclasses need to
+/// register themselves to [SelectionRegistrar]s. Use the
+/// [SelectionContainer.maybeOf] to get the the selection registrar, and mix
+/// the [SelectionRegisteree] to subscribe to the [SelectionRegistrar]
+/// automatically.
+///
+/// The [Selectable] also paints [LayerLink]s of selection handles in a
+/// mobile application. Use the [HandleLayersPainterMixin] to automatically
+/// paint the leader layers based on the [SelectionGeometry] this selectable
+/// returns.
+///
+/// {@macro flutter.rendering.SelectionHandler}
+mixin Selectable on RenderBox implements SelectionHandler { }
+
+/// A mixin to auto register the mixer to the [registrar].
+///
+/// To use this mixin, the subclasses need to set the [registrar] to the
+/// [SelectionRegistrar] it wants to register to and set the value of
+/// [hasSelectableContent]. This mixin only registers itself with the
+/// [registrar] if there is selectable content in this [Selectable].
+mixin SelectionRegisteree on Selectable {
+
+  /// The [SelectionRegistrar] this [Selectable] will be or is registered to.
+  SelectionRegistrar? get registrar => _registrar;
+  SelectionRegistrar? _registrar;
+  set registrar(SelectionRegistrar? value) {
+    if (value == _registrar)
+      return;
+    _removeSelectionRegistrarSubscription();
+    _registrar = value;
+    _updateSelectionRegistrarSubscription();
+  }
+
+  /// Whether there is selectable content in the selectable.
+  ///
+  /// Modify the value of this [ValueNotifier] to notify this mixin about
+  /// content changes.
```

Comment:
How do you do that? Do you just switch it between false and true whenever the content changes? Seems ... odd.

Reply:
Or do you only change this when it changes from having content to not having content?